core_v14/tests/decode_extrinsics.rs

```diff
@@ -28,6 +28,38 @@ fn to_bytes(hex_str: &str) -> Vec<u8> {
 	hex::decode(hex_str).expect("valid bytes from hex")
 }
 
+fn unnamed_value(x: Vec<Value>) -> Value {
+	Value::Composite(value::Composite::Unnamed(x))
+}
+
+fn empty_value() -> Value {
+	unnamed_value(vec![])
+}
+
+fn singleton_value(x: Value) -> Value {
```

Comment:
Whats the idea behind storing `prim_u<int>_value()` in a singleton for the tests?

Reply:
had the same question!

Reply:
A singleton here is just a Composite value with only a single member. Some of the metadata types decode into such a thing (eg newtype wrapper structs), and so we need to do the same wrapping when comparing the `Value` we get back.